components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java

```diff
@@ -844,6 +848,53 @@ public Response getAttributes(@PathParam("siddhiAppName") String siddhiAppName,
                 ).build();
     }
 
+    /**
+     * Converts a given Siddhi App string to a specific JSON format for a graph that diagrammatically
+     * display's the Siddhi App to be generated in the Editor design view.
+     *
+     * @param siddhiAppBase64 The Siddhi App (encoded to Base64) to be converted to JSON
+     * @return The JSON result in a predefined format
+     */
+    @POST
+    @Path("/event-flow")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response constructEventFlowJsonString(String siddhiAppBase64) {
+        Response response;
+
+        try {
+            String siddhiAppString = new String(Base64.getDecoder().decode(siddhiAppBase64),
+                    Charset.defaultCharset());
+
+            SiddhiAppMap siddhiAppMap = new SiddhiAppMap(siddhiAppString);
+            EventFlow eventFlow = new EventFlow(siddhiAppMap);
+
+            // The 'Access-Control-Allow-Origin' header must be set to '*' as this might be accessed
+            // by other domains in the future.
+            response = Response.status(Response.Status.OK)
+                    .header("Access-Control-Allow-Origin", "*")
+                    .entity(eventFlow.getEventFlowJSON().toString())
+                    .build();
+        } catch (SiddhiAppCreationException e) {
+            // SiddhiAppCreationException is caught if the provided 'SiddhiAppString' contains any syntax errors.
```

Comment:
This comment seems redundant. Let the log message speak :speaker:

Reply:
A comment was requested to be written in the code review to explain why this exception is caught.

Reply:
thats becoz the log message is not informative, lets the log message tell the exception/error. 

Log messages should be descriptive and informative. They should tell what is the error, what process was failed due to the exception, what are the relevant parameters that causes this etc. And the Exceoption `e` should be passed to ERROR logs so that the stacktrace is logged which shows the execution path.